### PR TITLE
bump helm/v3 to 3.11.1 && bump containerd to 1.5.18 for security

### LINKED ISSUE #4849: Vulnerability of dependency "github.com/containerd/containerd"
Hello, we are a team researching the dependency management mechanism of Golang. During our analysis, we came across your project and noticed that it contains a vulnerability (CVE-2023-25153). In your project, the github.com/containerd/containerd package is being used at version github.com/containerd/containerd v1.5.16, but the patched version is v1.6.18. To fix the vulnerability, we recommend modifying the go.mod file to update the version to v1.6.18 or higher. Thank you for your attention to this matter.

### PR DESCRIPTION
**Pull request by Shelley-BaoYue**
**What type of PR is this?**

/kind bug
/kind cleanup

**What this PR does / why we need it**:

bump helm.sh/helm/v3 to v3.11.1 for https://github.com/advisories/GHSA-pwcw-6f5g-gxf8
bump github.com/containerd/containerd to v1.5.18 for https://github.com/advisories/GHSA-259w-8hf6-59c2

**Which issue(s) this PR fixes**:

Fixes #4849

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,48 +7,48 @@ require (
...
 	golang.org/x/time v0.0.0-20220210224613-90d013bbcef8
-	google.golang.org/grpc v1.43.0
-	google.golang.org/protobuf v1.27.1
-	helm.sh/helm/v3 v3.9.4
-	k8s.io/api v0.24.14
-	k8s.io/apiextensions-apiserver v0.24.14
-	k8s.io/apimachinery v0.24.14
-	k8s.io/apiserver v0.24.14
-	k8s.io/cli-runtime v0.24.14
-	k8s.io/client-go v0.24.14
```
Comment (fisherxu) [inline go.mod:42]: The kubernetes related versions also need to update?

Comment (fisherxu) [reply go.mod:42]: And it's better to split go mod in one commit, others in another :)

Comment (Shelley-BaoYue) [reply go.mod:42]: > The kubernetes related versions also need to update?

 Kubernetes versions are updated when bump helm to 3.11.1, but these are replaced to v1.24.4-kubeedge1 in the back part.

Comment (Shelley-BaoYue) [reply go.mod:42]: > And it's better to split go mod in one commit, others in another :)

done.

**Review by fisherxu — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**